governance/program/src/instruction.rs

```diff
@@ -970,24 +980,39 @@ pub fn create_proposal(
 pub fn add_signatory(
     program_id: &Pubkey,
     // Accounts
+    governance: &Pubkey,
     proposal: &Pubkey,
-    token_owner_record: &Pubkey,
-    governance_authority: &Pubkey,
+    permission: &AddSignatoryPermission,
```

Comment:
There is no really naming convention for the authority signing a given instruction and unfortunately `governance_authority` is a misnomer across the code base because the name implies it's an authority somehow related to the `Governance` account while   it's  in fact the authority signing on behalf of  `TokenOwner`.

We should make a separate PR to address the name change across all instructions but for this PR we could make it a little better without much refactoring. 

- Rename `permission` to `add_signatory_authority` to have a pattern of `INSTRUCTION_NAME_authority` to clearly indicate this is the instruction signing authority 
- Then `AddSignatoryPermission` can become 

```
pub enum AddSignatoryAuthority {
    /// Proposal owner can add optional signatories to a proposal
    ProposalOwner {
        /// Token owner record of the Proposal owner
        token_owner_record: Pubkey,
        /// Token owner or its delegate
        governance_authority: Pubkey,
    },
    /// Anyone can add signatories that are required by the governance to a proposal
    None,
}
```

Reply:
Resolved. That's much better.